wire-ios/Wire-iOS/Sources/UserInterface/Settings/AccentColorPickerController.swift

```diff
@@ -16,217 +16,38 @@
     }
 
-    override func viewDidLoad() {
-        super.viewDidLoad()
-        setupViews()
-        createConstraints()
-    }
-
-    private func setupViews() {
-        view.backgroundColor = SemanticColors.View.backgroundDefault
-        view.addSubview(tableView)
+    init(selectedAccentColor: Binding<AccentColor?>) {
+        let allAccentColors = AccentColor.allCases
+        var initialSelectedColor: AccentColor?
 
-        tableView.register(PickerCell.self, forCellReuseIdentifier: PickerCell.reuseIdentifier)
-        tableView.delegate = self
-        tableView.dataSource = self
-        tableView.separatorStyle = .none
-        self.navigationItem.rightBarButtonItem = navigationController?.closeItem()
-    }
-
-    private func createConstraints() {
-        tableView.translatesAutoresizingMaskIntoConstraints = false
-
-        NSLayoutConstraint.activate([
-            tableView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
-            tableView.heightAnchor.constraint(equalToConstant: Self.rowHeight * CGFloat(colors.count)),
-            tableView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
-            tableView.trailingAnchor.constraint(equalTo: view.trailingAnchor)
-        ])
-    }
-
-    fileprivate final class PickerCell: UITableViewCell {
-        fileprivate let checkmarkView = UIImageView()
-        fileprivate let colorView = UIView()
-        fileprivate let colorNameLabel: UILabel = {
-            let label = UILabel()
-            label.font = .normalLightFont
-            label.textColor = SemanticColors.Label.textDefault
-            return label
-        }()
-
-        override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
-            super.init(style: style, reuseIdentifier: reuseIdentifier)
-            setupViews()
-            createConstraints()
-        }
-
-        @available(*, unavailable)
-        required init?(coder aDecoder: NSCoder) {
-            fatalError("init(coder:) has not been implemented")
+        if let bindingValue = selectedAccentColor.wrappedValue {
+            // If the binding has an initial value, use it
+            initialSelectedColor = bindingValue
+        } else if let firstColor = allAccentColors.first {
+            // If no initial value, use the first available color as a default
+            initialSelectedColor = firstColor
         }
 
-        var color: AccentColor? {
-            didSet {
-                if let color {
-                    colorView.backgroundColor = color.uiColor
+        super.init(rootView: ColorPickerView(selectedColor: initialSelectedColor, colors: allAccentColors, onColorSelect: { selectedColor in
+            selectedAccentColor.wrappedValue = selectedColor
+            if let colorIndex = allAccentColors.firstIndex(of: selectedColor) {
+                ZMUserSession.shared()?.perform {
+                    ZMUser.selfUser()?.accentColorValue = allAccentColors[colorIndex].rawValue
```

Comment:
Can we avoid the closure and just pass the binding to the picker?

Reply:
@johnxnguyen Not sure I follow. Can you please expand on the idea?

Reply:
I think the code changed, but when I reviewed it there was a `Binding` available that you could have passed to the picker. 

`AccentColorPicker` is a control, and typically (according to Apple) controls use bindings to get and set values rather than closures. See for instance the [`init` for `ColorPicker`](https://developer.apple.com/documentation/swiftui/colorpicker/init(selection:supportsopacity:label:)-1ijcv):

```swift
init(
    selection: Binding<Color>,
    supportsOpacity: Bool = true,
    @ViewBuilder label: () -> Label
)
```

I think it would be nice to stick with this convention and have a similar init for `AccentColorPicker`:

```swift
init(selection: Binding<AccentColor>)
```

Reply:
I think it depends a bit what one wants to achieve in SwiftUI:
- Button for example just calls a closure for [tab action](https://developer.apple.com/documentation/swiftui/button/init(_:action:)-1asy).
- To communicate a value out of the view a `Binding` seems the right choice.

So in this case I agree we can use a `Binding` here for the view itself. In our case with this architecture to go through a factory currently, we "misuse" the `AccentColorPickerController` as wrapper for business logic, so I suggest the binding can be created there and pass and receive the value to/from the view.

Reply:
I added a comment why Binding is hard to use, so I'm marking this comment as resolved.